state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -76,6 +79,13 @@ pub enum BoostStatus<ChainAmount> {
 	NotBoosted,
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, Encode, Decode, TypeInfo, Default)]
+pub enum Prewitnessed {
```

Comment:
As a suggestion, maybe we could rename this to `BoostRejected`? It helped me understand the code when thinking about it this way. Right now it is possible that you prewitnessed a tx, then marked it as tainted, but the status will still be `Prewitnessed::No`, correct?